includes/functions.php

```diff
@@ -588,6 +588,67 @@ function pmpro_getLevelsExpiration( &$levels ) {
 	return $expiration_text;
 }
 
+/**
+ * Get the text to display a membership's expiration date.
+ * @since 3.0
+ * @param int $enddate The expiration date timestamp. 
+ * @param mixed $level A level object or id (optional) to pass to filters.
+ * @param mixed $user A user object or id (optional) to pass to filters.
+ * @return string The expiration date text.
+ */
+function pmpro_get_membership_expiration_text( $enddate, $level = null, $user = null ) {	
+	if ( empty( $enddate ) ) {
+		$text = esc_html__( 'Never', 'paid-memberships-pro' );
+	} elseif( is_numeric( $enddate ) ) {		
+		/**
+		 * Filter to include the expiration time with expiration date
+		 * @param bool $pmpro_show_time_on_expiration_date Show the expiration time with expiration date
+		 * @param int $enddate The expiration date timestamp.
+		 * @param mixed $level A level object or id passed into this function.
+		 * @param mixed $user A user object or id passed into this function.
+		 * @return bool $pmpro_show_time_on_expiration_date Whether to show the expiration time with expiration date.
+		 * @since 3.0 Now passes the $enddate, $level, and $user.
+		 * 
+		 * Used in adminpages/member-edit/pmpro-class-member-edit-panel-memberships.php
+		 */
+		$show_time = apply_filters( 'pmpro_show_time_on_expiration_date', false, $enddate, $level, $user );
+		
+		// Convert enddate timestamp to a date string.
+		if ( $show_time ) {
+			$text = sprintf(
+				// translators: %1$s is the date and %2$s is the time.
+				esc_html__( '%1$s at %2$s', 'paid-memberships-pro' ),
+				date_i18n( get_option( 'date_format'), $enddate ),
+				date_i18n( get_option( 'time_format'), $enddate )
+			);
+		} else {
+			$text = date_i18n( get_option( 'date_format' ), $enddate );
+		}		
```

Comment:
The tough part here is that custom code and add ons filtering end date text would have to code support for this filter as well.

Instead of having this filter and logic, could we instead use the new filter for this function whenever we want to change the format of the function's output?

Reply:
Nice. Moved the filter outside the conditional so it's always used set up. Also passing it to the main/bottom filter.